lib/knapsack_pro/repository_adapters/git_adapter.rb

```diff
@@ -14,8 +14,40 @@ def branches
         str_branches.split("\n")
       end
 
+      def commit_authors
+        authors = git_commit_authors
+          .split("\n")
+          .map { |line| line.strip }
+          .map { |line| line.split("\t") }
+          .map do |commits, author|
+            { commits: commits.to_i, author: KnapsackPro::MaskString.call(author) }
+          end
+
+        raise if authors.empty?
+
+        authors
+      rescue Exception
+        []
+      end
+
+      def build_author
+        author = KnapsackPro::MaskString.call(git_build_author.strip)
+        raise if author.empty?
+        author
+      rescue Exception
+        "no git <no.git@example.com>"
+      end
+
       private
 
+      def git_commit_authors
+        `git fetch --shallow-since "1 month ago" >/dev/null 2>&1 && git shortlog --summary --email --since "one month ago"`
```

Comment:
If you run knapsack pro locally on your machine, you call `--shallow-since` command and when I want to use `git blame`, I only see commits from the last month.

I undo changes by using the following command:

```
git fetch --unshallow || git fetch --all
```

I guess someone might not like that we mess up with their local repository when they try to run Knapsack Pro locally.

Reply:
Thanks for the report.

Reply:
For reference: The above issue will be solved when we run `--shallow-since` only in the context of the CI. This PR solves it:

* https://github.com/KnapsackPro/knapsack_pro-ruby/pull/209